eventcrypto.go

```diff
@@ -0,0 +1,161 @@
+package gomatrixserverlib
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"golang.org/x/crypto/ed25519"
+)
+
+// AddContentHashesToEvent sets the "hashes" key of the event with a SHA-256 hash of the unredacted event content.
+// This hash is used to detect whether the unredacted content of the event is valid.
```

Comment:
Also, this comment is misleading. AFAICT this has nothing to do with `event.content`, it's adding hashes for the entire event. Am I wrong?